commands/bmc.vegman

```diff
@@ -353,3 +353,86 @@ function cmd_reboot {
   echo "Reboot the BMC..."
   reboot ${force}
 }
+
+# @doc cmd_remoteimage
+# Access to the remote side file
+function cmd_remoteimage {
+  subcommand "$@"
+}
+
+# @sudo cmd_remoteimage_connect admin
+# @doc cmd_remoteimage_connect
+# Connect to the remote server via the specified protocol
+#   --nbd <0..10>     use NBD protocol to connect the remote server with specified NBD device number
+#   address[:port]    remote server address
+function cmd_remoteimage_connect {
+    local nbd_device=
+    local remote_point=""
+    local address=""
+    local port=""
+
+    while [[ $# -gt 0 ]]; do
+      case "${1}" in
+        --nbd)
+          if [[ $# -lt 2 ]]; then
+            echo "The NBD device number shall be specified" >&2
+            return 1
+          fi
+          nbd_device="/dev/nbd${2}"
+          shift
+          ;;
+        *)
+          if [[ -n "${remote_point}" ]]; then
+            abort_badarg "${1}"
+          else
+            remote_point="${1}"
+          fi
+          address="${remote_point%:*}"
+          port="${remote_point#*:}"
+          ;;
+      esac
+      shift
+    done
+
```

Comment:
I believe, you should test if remote_point empty here

Reply:
Why? The `remote_point` might contains only the IP-address without a specified port.
If `address` was not specified, only this leads to failure.

Reply:
Additionally, if we add here checking the remote_point, 
this will lead to a double-coding of failure message declaring.

Reply:
what happens in case used omit second parameter? 
As far as I see, all of variables will be empty, isn't?
    local remote_point=""
    local address=""
    local port=""
BTW, same for nbd_device - you don't ensure if it set.

Reply:
Yes, it may be omitted, and all the above variables will be empty.
In that case, the port will be filled from the default value (10809),
and the address variable state empty.
The script goes fail and prints message error at line #401.

The `--nbd` second parameter will be checked at the retrieving loop, and if it is omitted,
then all other parameters will be shifted.
And the result of that will be an error of specifying `remote server address.`

Reply:
ok, I agree about remote_point
but is it allowed to omit "${nbd_device}" parameter for nbd-client?

Reply:
Well, the `${nbd_device}` won't be empty; check is performed in line #424


```
--nbd)
        if [[ $# -lt 2 ]]; then
          echo "The NBD device number shall be specified" >&2      
```

And the same check implemented for the cmd_remoteimage_disconnect